contracts/RewardController.sol

```diff
@@ -61,21 +117,189 @@ contract RewardController is Ownable {
+            lastProcessedTotalAllocPoint: 0,
+            rewardPerShare: 0,
+            totalShares: 0,
+            allocPoint: 0
+        }));
+        _setAllocPoint(poolId, _allocPoint);
+        setPoolsLastProcessedTotalAllocPoint(poolId);
+    }
+
+    /**
+    * @notice set the shares of users in a pool
+    * only calleable by the owner, and only when a pool is not initialized
+    * This function is used for migrating older pool data to this new contract
+    * (and this function can be removed in the next upgrade, because the current version is upgradeable)
+    */
+    function setShares(
+        uint256 _pid,
+        uint256 _rewardPerShare,
+        address[] memory _accounts,
+        uint256[] memory _shares,
+        uint256[] memory _rewardDebts)
+    external onlyVaults {
+        PoolInfo storage pool = poolInfos[_pid];
+
+        pool.rewardPerShare = _rewardPerShare;
+
+        for (uint256 i = 0; i < _accounts.length; i++) {
+            userInfo[_pid][_accounts[i]] = UserInfo({
+                shares: _shares[i],
+                rewardDebt: _rewardDebts[i]
+            });
+            pool.totalShares += _shares[i];
+        }
+    }
+
+    /**
+     * @notice Called by owner to set reward multipliers
+     * @param _rewardPerEpoch reward multipliers
+    */
+    function setRewardPerEpoch(uint256[24] memory _rewardPerEpoch) external onlyOwner {
+        rewardPerEpoch = _rewardPerEpoch;
+        emit SetRewardPerEpoch(_rewardPerEpoch);
+    }
+
+    function setAllocPoint(uint256 _pid, uint256 _allocPoint) external onlyOwner {
+        updatePool(_pid);
+        _setAllocPoint(_pid, _allocPoint);
+    }
+
+    /**
+     * @notice Transfer to the sender their pending share of rewards.
+     * @param _pid The pool id
+     */
+    function claimReward(uint256 _pid) external {
+        updatePool(_pid);
+
+        UserInfo storage user = userInfo[_pid][msg.sender];
+        uint256 rewardPerShare = poolInfos[_pid].rewardPerShare;
+        if (user.shares > 0) {
+            uint256 pending = user.shares * rewardPerShare / 1e12 - user.rewardDebt;
+            if (pending > 0) {
+                user.rewardDebt = user.shares * rewardPerShare / 1e12;
+                safeTransferReward(msg.sender, pending, _pid);
+            }
+        }
+
+        emit ClaimReward(_pid);
     }
 
+    /**
+     * @notice add reward tokens to the hatVaults contrac, to be distributed as rewards
+     * The sender of the transaction must have approved the spend before calling this function
+     * @param _amount amount of rewardToken to add
+    */
+    function depositReward(uint256 _amount) external {
+        uint256 balanceBefore = rewardToken.balanceOf(address(this));
+        rewardToken.safeTransferFrom(msg.sender, address(this), _amount);
+        uint256 rewardTokenReceived = rewardToken.balanceOf(address(this)) - balanceBefore;
+        rewardAvailable += rewardTokenReceived;
```

Comment:
so we had the "rewardAvailable" logic because we needed to distinguish bhe balance of the hats token that is for rewards from those that we in the hats vault. That is not necessary anymore so I think we can remove all rewardAvailable stuff and just use the contract balance